From d02d17986e07d1eb4c25b00b445ab84e41fbfbb5 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Wed, 9 Nov 2022 19:47:34 +0800
Subject: [PATCH 0003/1398] [Huawei] Revert "[Huawei] block: Fix UAF in
 bd_link_disk_holder()"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187941

--------------------------------

This reverts commit b4d9c5268b6b2e4a0ee2d1629b714e18328034ff.

Official solution will be applied to mainline, and this solution can't
fix uaf for 'bd_holder_dir' thoroughly. hence revert this temporary
solution. Officail solution will be backported in the next patch.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 fs/block_dev.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/fs/block_dev.c b/fs/block_dev.c
index 069a323089f9..1917c78c5552 100644
--- a/fs/block_dev.c
+++ b/fs/block_dev.c
@@ -1632,7 +1632,6 @@ static int __blkdev_get(struct block_device *bdev, fmode_t mode, void *holder,
 		}
 	}
 	bdev->bd_openers++;
-	kobject_get(bdev->bd_part->holder_dir);
 	if (for_part)
 		bdev->bd_part_count++;
 
@@ -1902,7 +1901,6 @@ static void __blkdev_put(struct block_device *bdev, fmode_t mode, int for_part)
 	}
 #endif
 
-	kobject_put(bdev->bd_part->holder_dir);
 	if (!--bdev->bd_openers) {
 		WARN_ON_ONCE(bdev->bd_holders);
 		sync_blockdev(bdev);
-- 
2.17.1

